tests/unit/core/CMakeLists.txt

```diff
@@ -1,20 +1,20 @@
-cmake_minimum_required(VERSION 3.10) 
-#project(tests-unit VERSION 0.1)
-
-set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED True)
-
-file(STRINGS targets.txt TARGETS)
+set(TEST_PREFIX "${TEST_PREFIX}-core")
+# Find all of the .cpp files in this directory
+file(GLOB TARGET_SOURCES CONFIGURE_DEPENDS RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.cpp)
 
 set(MAIN_SOURCE_DIR ../${MAIN_SOURCE_DIR})
 
-foreach(TARGET IN LISTS TARGETS)
-  add_executable(tests-unit-core-${TARGET} ${TARGET}.cpp)
+foreach(TARGET_SOURCE IN LISTS TARGET_SOURCES)
+  string(REPLACE ".cpp" "" TARGET_NAME ${TARGET_SOURCE})
+  set(EXE_NAME "${TEST_PREFIX}-${TARGET_NAME}")
+  message(STATUS "Building tests for: ${EXE_NAME}")
+  add_executable(${EXE_NAME} ${TARGET_SOURCE})
 
-  target_include_directories(tests-unit-core-${TARGET}
-    PRIVATE ${MAIN_SOURCE_DIR}
+  target_include_directories(${EXE_NAME}
+    PRIVATE ${CMAKE_SOURCE_DIR}/source
     )
-  target_link_libraries(tests-unit-core-${TARGET}
+  target_link_libraries(${EXE_NAME}
     PRIVATE Catch2::Catch2WithMain
     )
+  add_test(NAME ${EXE_NAME} COMMAND ${EXE_NAME})
```

Comment:
we have to add in `WORKING_DIRECTORY ${CMAKE_BINARY_DIR}` so that files in assets can be picked up properly

example test
`add_test(NAME tests-unit-gp_agents-ctest-${TARGET} COMMAND tests-unit-gp_agents-${TARGET} WORKING_DIRECTORY ${CMAKE_BINARY_DIR})`